docs/faq.rst

```diff
@@ -0,0 +1,211 @@
+process multiple messages.
+
+For example, if your handler looks like this::
+
+    async def handler(websocket, path):
+        print(websocket.recv())
+
+change it like this::
+
+    async def handler(websocket, path):
+        async for message in websocket:
+            print(message)
+
+*Don't feel bad if this happens to you — it's the most common question in
+websockets' issue tracker :-)*
+
+Why can only one client connect at a time?
+..........................................
+
+Your connection handler blocks the event loop. Look for blocking calls.
+Any call that may take some time must be asynchronous.
+
+For example, if you have::
+
+    async def handler(websocket, path):
+        time.sleep(1)
+
+change it to::
+
+    async def handler(websocket, path):
+        await asyncio.sleep(1)
+
+This is part of learning asyncio. It isn't specific to websockets.
+
+See also Python's documentation about `running blocking code`_.
+
+.. _running blocking code: https://docs.python.org/3/library/asyncio-dev.html#running-blocking-code
+
+How do I get access HTTP headers, for example cookies?
+......................................................
+
+To access HTTP headers during the WebSocket handshake, you can override
+:attr:`~server.WebSocketServerProtocol.process_request`::
+
+    async def process_request(self, path, request_headers):
+        cookies = request_header["Cookie"]
+
+See
+
+Once the connection is established, they're available in
+:attr:`~protocol.WebSocketServerProtocol.request_headers`::
+
+    async def handler(websocket, path):
+        cookies = websocket.request_headers["Cookie"]
+
+How do I get the IP address of the client connecting to my server?
+..................................................................
+
+It's available in :attr:`~protocol.WebSocketCommonProtocol.remote_address`::
+
+    async def handler(websocket, path):
+        remote_ip = websocket.remote_address[0]
+
+How do I set which IP addresses my server listens to?
+.....................................................
+
+Look at the ``host`` argument of :meth:`~asyncio.loop.create_server`.
+
+:func:`serve` accepts the same arguments as
+:meth:`~asyncio.loop.create_server`.
+
+How do I close a connection properly?
+.....................................
+
+websockets takes care of closing the connection when the handler exits.
+
+How do I run a HTTP server and WebSocket server on the same port?
+.................................................................
+
+This isn't supported.
+
+Providing a HTTP server is out of scope for websockets. It only aims at
+providing a WebSocket server.
+
+There's limited support for returning HTTP responses with the
+:attr:`~server.WebSocketServerProtocol.process_request` hook.
+If you need more, pick a HTTP server and run it separately.
+
+Client side
+-----------
+
+How do I close a connection properly?
+.....................................
+
+Use :func:`connect` as a context manager and trust websockets::
```

Comment:
Is that sentence incomplete? *Trust websockets* regarding?

Reply:
Implicitly, "to do the right thing". I'll phrase it differently.